pypsa/plot.py

```diff
@@ -247,27 +256,40 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
         bus_collection = ax.scatter(x, y, c=c, s=s, cmap=bus_cmap, edgecolor='face')
 
     def as_branch_series(ser):
+        # ensure that this function always return a multiindexed series
         if isinstance(ser, dict) and set(ser).issubset(branch_components):
-            return pd.Series(ser)
-        elif isinstance(ser, pd.Series):
-            if isinstance(ser.index, pd.MultiIndex):
-                return ser
-            index = ser.index
-            ser = ser.values
+            return network.branches().loc[branch_components] \
+                          .join(pd.Series(ser, name='value') \
+                          .rename_axis('component')).value
```

Comment:
Can you get this also working without creating a copy of all static branch attributes just to compose the relatively branches multiindex. That is so wasteful.

Maybe something along the lines of
```python
pd.concat({c.name: pd.Series(s, index=c.df.index)
           for c, s in zip(network.iterate_components(ser.keys()), ser.values())})
```

Reply:
yes great, I searched for something like this